general/arrays_by_name.hpp

```diff
@@ -279,8 +279,9 @@ void ArraysByName<T>::Load(std::istream &in)
          ArrayName = ArrayLine.substr(0,q1-1);
       }
 
-      // Ignore the remainder of the line which may contain explanatory comments
-      data[ArrayName].Load(in, 0);
+      std::istringstream iss(ArrayLine.substr(q1 + 1));
+
+      data[ArrayName].Load(iss, 0);
```

Comment:
This implementation does not allow the numbers after the name to be on new lines -- we typically allow arbitrary white space characters between entries -- it will be good to allow that here as well.

Reply:
> we typically allow arbitrary white space characters between entries 

understood. I will setup a few gtests to cover those requirements and update this PR within a week.